ingestor/storage/crds.go

```
@@ -164,3 +167,61 @@ func ConvertToTypedList(objects []client.Object, list client.ObjectList) error {
 	itemsField.Set(newSlice)
 	return nil
 }
+
+// CheckOwnership returns true if the object is owned by the controller or if the contoller has successfully
+// claimed ownership of the resource.
+func CheckOwnership(ctx context.Context, obj client.Object, ctrlClient client.Client, ownerId string) bool {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	owner, owned := annotations[adxmonv1.ControllerOwnerKey]
+	lastUpdated, hasLastUpdated := annotations[adxmonv1.LastUpdatedKey]
+
+	// If unowned, claim it
+	if !owned {
+		if logger.IsDebug() {
+			logger.Debugf("Claiming unowned resource %s/%s", obj.GetNamespace(), obj.GetName())
+		}
+		annotations[adxmonv1.ControllerOwnerKey] = ownerId
+		annotations[adxmonv1.LastUpdatedKey] = metav1.Now().Format(time.RFC3339)
+		obj.SetAnnotations(annotations)
+		if err := ctrlClient.Update(ctx, obj); err == nil {
+			if logger.IsDebug() {
+				logger.Debugf("Claimed ownership of %s/%s: %v", obj.GetNamespace(), obj.GetName(), err)
+			}
+			owner = ownerId
+		}
```

Handle err != nil case?

---

Errors here are expected due to conflicting ResourceVersions. As two ingestor instances try to take ownership of the same object, one will become the owner and the other will generate an error, which can be ignored.

---

I'm just concerned that we might be swallowing an different error that we should log.  Would it make sense to explicitly check for the conflict error and ignore, but log the other cases that we would not expect?